Iteration 3

aliciatay-zls opened the pull request:
Complete to-do list by mentor:

- [x] Load only selected columns into db
- [x] `brands` table, Foreign Key for sample data
- [x] `BrandName` column in `outlets` table (join query)
- [x] AJAX call from index.html to `outlets` route
- [ ] Improve user's readability of distances displayed
- [ ] Add new sample data: Fairprice
- [x] Enable auto-reloading of REST API whenever there is a change in `rest_api.js`

aliciatay-zls committed:
Remove index column, change to insert selected columns only

miccheng commented on mysql_scripts/setup.sql:
```diff
@@ -3,12 +3,11 @@ CREATE DATABASE `locationsDB`;
 USE `locationsDB`;
 
 CREATE TABLE `outlets` (
-    `OutletId` INT unsigned NOT NULL AUTO_INCREMENT,
```
I feel you should still have an ID column.

miccheng commented on mysql_scripts/setup.sql:
```diff
@@ -3,12 +3,11 @@ CREATE DATABASE `locationsDB`;
     `Closing` varchar(100) NOT NULL,
-    PRIMARY KEY (`OutletId`)
-) ENGINE=InnoDB AUTO_INCREMENT=5 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+    PRIMARY KEY (`OutletName`)
```
OutletName might also be duplicated - eg. KFC and NTUC might have a "Toa Payoh" branch.

aliciatay-zls replied:
Sorry about that, I misunderstood. Fixed, thank you.

miccheng reviewed (changes requested):
Consider my recommendations.

aliciatay-zls committed:
Remove redundant ID column in generated .csv

aliciatay-zls commented:
> Consider my recommendations.

Sorry about that, I misunderstood. Fixed, thank you.

aliciatay-zls committed:
Make basic ajax call to outlets route (does nothing)

aliciatay-zls committed:
Add `brands table`, Foreign Key for sample data

- add ability to generate .csv file for brands in parser.js
- fix `load data infile` issues in db.sql

aliciatay-zls committed:
Add BrandName col in `outlets` table, create join query

- remove dead code in db.sql

aliciatay-zls committed:
Implement join query in rest_api.js, add section title to sent JSON

miccheng commented on html/index.html:
```diff
@@ -26,13 +27,22 @@
+                var params = {
+                    searchWord: $("input[name=searchWord]").val()
+                };
+                $.post("/outlets", params, function(res, status) {
```
I think before we complete the search function, maybe we use the `GET /outlets` first? So at least we can show the results first before thinking about parsing the search string.

aliciatay-zls replied:
Okay, will try that out now. Thanks Mr Cheng!

aliciatay-zls committed:
Make AJAX call from index.html to outlets route

aliciatay-zls committed:
Make AJAX call from index.html to outlets route

aliciatay-zls committed:
Merge branch 'iteration-3' of https://github.com/aliciatay-zls/NearMe into iteration-3

aliciatay-zls committed:
Enable auto-reloading of rest_api.js with file changes

miccheng commented on html/index.html:
```diff
@@ -33,16 +33,55 @@
-        $(document).ready(function(){
-            $(":submit").click(function(){
+        // Endpoint: http://localhost:3000/outlets?searchWord=kfc&currentLatitude=1.3730702&currentLongitude=103.9028103
+        // NOTE: code not working
```
https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault <-- maybe you need to do this to prevent the form from posting.

aliciatay-zls committed:
Finish up enable auto-reloading of rest_api.js with mentor

- add config in package.json

aliciatay-zls committed:
Modify index.html for url building

miccheng reviewed (approved):
LGTM
